src/rfc3986/_mixin.py

```diff
@@ -380,6 +404,6 @@ def copy_with(
         for key, value in list(attributes.items()):
             if value is misc.UseExisting:
                 del attributes[key]
-        uri = self._replace(**attributes)
+        uri: "uri.URIReference" = self._replace(**attributes)
```

Comment:
Couldn't this be done on an IRIReference too? Wouldn't this then be `uri: "URIMixin"`?

Reply:
I think `_Self` would be most accurate, but that would work too. I think I got frustrated here because of the lack of an encoding annotation for `URLMixin` and forgot in the moment this mixin is subclassed multiple times. I'll fix it.